### ISSUE TITLE
Tests fail with time zone EEST

### ISSUE THREAD

**Issue by akalfas** [adobe/zdw#12]
### Expected Behaviour
Tests should run irrespective of the local time zone settings.

### Actual Behaviour
mvn clean package yields

> should be able to read and match unconvertDWfile output from test-files *** FAILED ***
>   movie_tickets.zdw:
>   Row[121080][0][date_time / date_time]: "2018-03-25 0[4]:00:00" was not equal to "2018-03-25 0[3]:00:00" (ZDWStreamIteratorTest.scala:118)

#### Steps to Reproduce
Change time zone to EEST - rerun test.

**Comment by akalfas** [comment]
It has to do with day light saving in this particular case. 2018-03-25 03:00:00 is 2018-03-25 04:00:00 with the EEST timezone.

**Comment by akalfas** [comment]
@clubycoder I think that in order to make this behave correctly time zone needs to be preserved in the sql files and the writer needs to serialize that information as well such that readers can correctly reason about the date. 

wdyt ?

**Comment by scotthunt** [comment]
I believe that dates are stored as plain strings in ZDW.  No conversion should be done by the reader (which it sounds like is broken in the scala reader?)
As far as I could tell, the C++ reader works fine in this regard.

**Comment by akalfas** [comment]
Reading the writer code I've noticed that https://github.com/adobe/zdw/blob/master/cplusplus/zdw_column_type_constants.h#L19 is used for handling dates, reading https://dev.mysql.com/doc/refman/5.7/en/datetime.html sounds like a mysql datetime type is used as original format - and that does not contain a time zone information. W/o that information no date will be ever parsed correctly unless some additional assumptions are made - i.e. TZ is always UTC.

**Comment by mrimer** [comment]
@akalfas , I concur that users of MySQL datetime columns need to maintain TZ context in order to interpret such values correctly.  This is a fundamental issue with the use of MySQL datetime and not any particular compression format, per se.  The use of date_time is included in the movie_tickets sample file for illustrative purposes, and its inclusion in the repo is not attempting to be prescriptive in the user's managing of time information.  If MySQL users need TZ context embedded in column values, a different column type other than datetime should be used, e.g., timestamp.

That is, ZDW works with any generic string type that you may choose to store in it.  You don't need to use a MySQL datetime column type to handle dates in your raw storage data in conjunction with ZDW.

**Comment by clubycoder** [comment]
Right now the Scala reader should be parsing the date-time type fields into JVM localtime Date objects - https://github.com/adobe/zdw/blob/master/format/src/main/scala/com/adobe/analytics/zdw/format/ZDWBlock.scala#L60 - @carrus85 & @mrimer how does the C++ implementation deal with date-time parsing?  Or do we never parse them in the C++ implementation?  Do we just leave them as strings after they have been written?

**Comment by mrimer** [comment]
@clubycoder , correct: there should probably be no interpretation of string values in the ZDW unconverter, as ZDW is a compression format, not a semantic data container, per se, beyond retaining the knowledge that a column is of a specified type.  Any interpretation of values should probably be performed by the caller external to the unconversion of compressed data.

**Comment by clubycoder** [comment]
What I'm thinking right now is changing the parsing to always parse date-time as UTC vs using the JVM timezone.  I think that would be a good balance of treating the types as they are marked but making less assumptions.  A fixed UTC will be consistent and still force the user to deal with timezones but give them a head start abstracting the date string format from them.

How does that sound?

**Comment by mrimer** [comment]
I like that approach.  Maybe leave a hook for the user to optionally set the timezone for date-times?  Not sure that's necessary; you'd know best.

**Comment by clubycoder** [comment]
@mrimer , I like the idea of adding an option.  I'll think that through.

@akalfas , I was unable to reproduce the difference you saw forcing the JVM timezone EEST.  Can you try out the branch with my UTC fix - https://github.com/adobe/zdw/pull/14 - and see if that takes care of what you're seeing?

**Comment by akalfas** [comment]
@clubycoder - indeed setting the time zone via -Duser.timezone=EEST indeed does not yield the bug, doing it with -Duser.timezone=Europe/Bucharest does. TimeZone.getAvailableIDs() does not have EEST - therefore it defaults to GMT. 

The way that I've reproduced it initially was simply by running `mvn clean package` on my laptop and the odd thing is that setting the timezone to Europe/Bucharest and simply printing a date yields EEST - hence my misleading information about the timezone.

I've checked #14 and it looks good.
